src/rpc/handlers/AccountObjects.cpp

```diff
@@ -23,7 +24,141 @@ std::unordered_map<std::string, ripple::LedgerEntryType> types{
     {"escrow", ripple::ltESCROW},
     {"deposit_preauth", ripple::ltDEPOSIT_PREAUTH},
     {"check", ripple::ltCHECK},
-};
+    {"nft_page", ripple::ltNFTOKEN_PAGE},
+    {"nft_offer", ripple::ltNFTOKEN_OFFER}};
+
+Result
+doAccountNFTs(Context const& context)
+{
+    auto request = context.params;
+    boost::json::object response = {};
+
+    auto v = ledgerInfoFromRequest(context);
+    if (auto status = std::get_if<Status>(&v))
+        return *status;
+
+    auto lgrInfo = std::get<ripple::LedgerInfo>(v);
+
+    if (!request.contains("account"))
+        return Status{Error::rpcINVALID_PARAMS, "missingAccount"};
+
+    if (!request.at("account").is_string())
+        return Status{Error::rpcINVALID_PARAMS, "accountNotString"};
+
+    if (!request.at("account").is_string())
+        return Status{Error::rpcINVALID_PARAMS, "accountNotString"};
+
+    auto accountID =
+        accountFromStringStrict(request.at("account").as_string().c_str());
+
+    if (!accountID)
+        return Status{Error::rpcINVALID_PARAMS, "malformedAccount"};
+
+    // TODO: just check for existence without pulling
+    if (!context.backend->fetchLedgerObject(
+            ripple::keylet::account(*accountID).key,
+            lgrInfo.seq,
+            context.yield))
+        return Status{Error::rpcACT_NOT_FOUND, "accountNotFound"};
+
+    std::uint32_t limit = 200;
+    if (request.contains("limit"))
+    {
+        if (!request.at("limit").is_int64())
+            return Status{Error::rpcINVALID_PARAMS, "limitNotInt"};
+
+        limit = request.at("limit").as_int64();
+        if (limit <= 0)
+            return Status{Error::rpcINVALID_PARAMS, "limitNotPositive"};
+    }
+
+    ripple::uint256 cursor;
+    if (request.contains("marker"))
+    {
+        if (!request.at("marker").is_string())
+            return Status{Error::rpcINVALID_PARAMS, "markerNotString"};
+
+        if (!cursor.parseHex(request.at("marker").as_string().c_str()))
+            return Status{Error::rpcINVALID_PARAMS, "malformedCursor"};
+    }
+
+    auto const first = ripple::keylet::nftpage(
+        ripple::keylet::nftpage_min(*accountID), cursor);
+    auto const last = ripple::keylet::nftpage_max(*accountID);
+
+    auto const key =
+        context.backend
+            ->fetchSuccessorKey(first.key, lgrInfo.seq, context.yield)
+            .value_or(last.key);
```

Comment:
Do we know that `first.key` actually exists? `fetchSuccessorKey` will return nothing if `first.key` is not actually in the ledger, and the cache is not full.